src/WebJobs.Extensions.CosmosDB/Trigger/CosmosDBTriggerListener.cs

```diff
@@ -148,5 +181,253 @@ private void InitializeBuilder()
                     .WithObserverFactory(this);
             }
         }
+
+        private async Task<IRemainingWorkEstimator> GetWorkEstimatorAsync()
+        {
+            if (_workEstimatorBuilder == null)
+            {
+                _workEstimatorBuilder = new ChangeFeedProcessorBuilder()
+                    .WithHostName(this._hostName)
+                    .WithFeedDocumentClient(this._monitoredCosmosDBService.GetClient())
+                    .WithLeaseDocumentClient(this._leasesCosmosDBService.GetClient())
+                    .WithFeedCollection(this._monitorCollection)
+                    .WithLeaseCollection(this._leaseCollection)
+                    .WithProcessorOptions(this._processorOptions)
+                    .WithHealthMonitor(this._healthMonitor)
+                    .WithObserverFactory(this);
+            }
+
+            if (_workEstimator == null)
+            {
+                _workEstimator = await _workEstimatorBuilder.BuildEstimatorAsync();
+            }
+
+            return _workEstimator;
+        }
+
+        async Task<ScaleMetrics> IScaleMonitor.GetMetricsAsync()
+        {
+            IRemainingWorkEstimator estimator = await GetWorkEstimatorAsync();
+            return await GetMetricsAsync(estimator);
+        }
+
+        public async Task<CosmosDBTriggerMetrics> GetMetricsAsync()
+        {
+            IRemainingWorkEstimator estimator = await GetWorkEstimatorAsync();
+            return await GetMetricsAsync(estimator);
+        }
+
+        internal async Task<CosmosDBTriggerMetrics> GetMetricsAsync(IRemainingWorkEstimator workEstimator)
+        {
+            int partitionCount = 0;
+            long remainingWork = 0;
+            IReadOnlyList<RemainingPartitionWork> partitionWorkList = null;
+
+            try
+            {
+                partitionWorkList = await workEstimator.GetEstimatedRemainingWorkPerPartitionAsync();
+
+                partitionCount = partitionWorkList.Count;
+                remainingWork = partitionWorkList.Sum(item => item.RemainingWork);
+            }
+            catch (Microsoft.Azure.Documents.DocumentClientException e)
```

Comment:
Recommend having a single block for the DocumentClientException and InvalidOperationException handling since the logic is repeated. E.g.

```csharp
catch (Exception ex) when (ex is DocumentClientException || ex is InvalidOperationException )
```